app-games/src/main/java/com/aptoide/android/aptoidegames/promo_codes/PromoCodeBottomSheetContent.kt

```diff
@@ -0,0 +1,210 @@
+            contentDescription = null,
+            modifier = Modifier
+              .fillMaxWidth()
+              .padding(bottom = 24.dp)
+          )
+          Row(
+            verticalAlignment = Alignment.CenterVertically,
+            horizontalArrangement = Arrangement.spacedBy(8.dp),
+            modifier = Modifier.padding(bottom = 8.dp)
+          ) {
+            Image(
+              imageVector = getAppcoinsIconWithBackground(Palette.AppCoinsPink, Palette.White),
+              contentDescription = null,
+            )
+            Text(
+              text = stringResource(
+                id = R.string.promo_code_install_wallet_banner_title,
+                "20"
+              ), //TODO Hardcoded value (should come from backend in the future)
+              color = Palette.White,
+              style = AGTypography.InputsM
+            )
+          }
+          Text(
+            text = stringResource(id = R.string.promo_code_install_wallet_banner_body),
+            color = Palette.White,
+            style = AGTypography.Body,
+          )
+          AppItem(
+            app = walletApp,
+            onClick = {
+              navigate(
+                buildAppViewRouteByAppId(appId = walletApp.id, useStoreName = false)
+              )
+              dismiss()
+            },
+            modifier = Modifier.padding(bottom = 8.dp)
+          ) {
+            InstallViewShort(app = walletApp)
+          }
+          Divider(
+            modifier = Modifier
+              .fillMaxWidth()
+              .padding(bottom = 12.dp)
+          )
+        }
+
+        Text(
+          modifier = Modifier.padding(top = 4.dp),
+          text = stringResource(
+            id = R.string.promo_code_extra_title,
+            "10"
+          ), //TODO Hardcoded value (should come from backend in the future)
+          color = Palette.White,
+          style = AGTypography.Title
+        )
+        AppItem(
+          app = app,
+          modifier = Modifier.padding(bottom = 16.dp),
+          onClick = {
+            navigate(
+              buildAppViewRouteByAppId(appId = app.id, useStoreName = false)
+            )
+            dismiss()
+          }
+        ) {
+          if (!isWalletInstalled) {
+            TertiarySmallButton(
+              onClick = { showSnack(walletDisclaimer) },
+              title = stringResource(id = R.string.promo_code_use_button)
+            )
+          } else {
+            PrimarySmallButton(
+              onClick = {
+                val uri = Uri.parse("appcoins://promocode?promocode=$promoCode")
+                val intent = Intent(Intent.ACTION_VIEW, uri)
+                context.startActivity(intent)
+              },
+              title = stringResource(id = R.string.promo_code_use_button)
+            )
+          }
+        }
+      }
+
+      is AppUiState.Loading -> LoadingView()
+      is AppUiState.Error -> GenericErrorView(onRetryClick = walletReload)
```

Comment:
The error views are not shown correctly in the bottom sheet. It might be required to implement error screens for the bottom sheet